docs/gadget-devel/gadget-ebpf-api.md

```diff
@@ -471,6 +471,60 @@ A new field containing the flags set will be added to the event:
 - `ebpf.formatter.enum`: Name of the new field. If the annotation is not set and the source field name has a `_raw` suffix, the target name will be set to the source name without that suffix.
 - `ebpf.formatter.bitfield.separator`: Separator used. Defaults to `|`.
 
+## Flexible Strings
+
+Some gadgets may need to use strings whose maximum size can be configured at
+runtime to save memory. For that, gadgets can use the `gadget_flex_string`
+type:
+
+```c
+#ifndef BPF_NO_PRESERVE_ACCESS_INDEX
+#pragma clang attribute push(__attribute__((preserve_access_index)), \
+			     apply_to = record)
+#endif
+
+struct event {
+	gadget_flex_string mystring[MAX_SIZE];
+};
+
+#ifndef BPF_NO_PRESERVE_ACCESS_INDEX
+#pragma clang attribute pop
+#endif
+```
+
+:::warning
+
+The `gadget_flex_string` fields must be on the root of the event structure.
+
+:::
+
+The maximum size of the string can be set by the user at runtime using the
+`<fieldName>-size` parameter. For example, to set the maximum size of `mystring`
+to 64 bytes, the user can run:
```

Comment:
Sounds like this could become ambiguous when multiple structs use the same field name for gadget_flex_string. Now that we have the custom params: what about using an annotation instead (probably leads to a hen/egg problem, though)?

Reply:
Yes - something like `--long-names` or `--short` as a parameter that would set the annotation to on that field. We keep the flexibility of resizing freely (using the annotations, for advanced users) while still getting a good UX for the normal user.